cl/corpus_importer/management/commands/columbia_merge.py

```diff
@@ -0,0 +1,865 @@
+        # id: 7643871 we have the posture and the opinion text but in the xml file we
+        # only have the opinion text, cosine_sim: 0.1639075094124459 and
+        # percent_match: 73)
+        percent_match = compare_documents(file_opinion, cl_opinion)
+
+        # Sometimes one algorithm performs better than the other, this is due to some
+        # additional text, such as editor's notes, or the author, page number or posture
+        # added to the opinion
+        if cosine_sim < 0.60 and percent_match < 60:
+            continue
+
+        matches[i] = j
+
+    # Key is opinion position from file, Value is opinion position from cl opinion
+    # e.g. matches {0: 1, 1: 2} 0 is file opinion and 1 in cl opinion, 1 is file
+    # opinion and 2 is cl opinion
+    return matches
+
+
+def map_and_merge_opinions(
+    cluster_id: int,
+    cl_cleaned_opinions: list[dict],
+    columbia_opinions: list[dict],
+) -> None:
+    """Map and merge opinion data
+    :param cluster_id: Cluster id
+    :param cl_cleaned_opinions: list of cl opinions
+    :param columbia_opinions: list of columbia opinions from file
+    """
+
+    if len(columbia_opinions) == len(cl_cleaned_opinions):
+        # We need that both list to be cleaned, so we can have a more accurate match
+        matches = match_text_lists(
+            [
+                clean_opinion_content(op["opinion"], is_harvard=False)
+                for op in columbia_opinions
+            ],
+            [op.get("opinion") for op in cl_cleaned_opinions],
+        )
+        if len(matches) == len(columbia_opinions):
+            update_matching_opinions(
+                matches, cl_cleaned_opinions, columbia_opinions
+            )
+        else:
+            raise OpinionMatchingException("Failed to match opinions")
+
+    elif (
+        len(columbia_opinions) > len(cl_cleaned_opinions)
+        and len(cl_cleaned_opinions) == 1
+    ):
+        for op in columbia_opinions:
+            opinion_type = op.get("type")
+            file = op.get("file")
+            if not opinion_type:
+                raise OpinionTypeException(
+                    f"Opinion type unknown: {op.get('type')} found in: {file}"
+                )
+            author = op.get("byline")
+
+            converted_text = convert_columbia_html(op["opinion"], op["order"])
+
+            Opinion.objects.create(
+                html_columbia=converted_text,
+                cluster_id=cluster_id,
+                type=opinion_type,
+                author_str=titlecase(find_just_name(author.strip(":")))
+                if author
+                else "",
+            )
+
+            logger.info(f"Opinion created for cluster: {cluster_id}")
+
+    else:
+        # Skip creating new opinion cluster due to differences between data
+        # NOTE: this may happen because some opinions were incorrectly combined when
+        # imported with the old columbia importer, or we have combined opinions
```

Comment:
this comment seems incorrect.  we shouldn't be handling anything that was already imported.

Reply:
i made a slight change in the comment to it makes more sense